extensions/frontend-tiles/src/MeshExportService.ts

```diff
@@ -0,0 +1,186 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+import { AccessToken, Logger } from "@itwin/core-bentley";
+import { IModelApp, IModelConnection } from "@itwin/core-frontend";
+import { loggerCategory } from "./LoggerCategory";
+
+/** A function that can provide the base URL where a tileset representing all of the spatial models in a given iModel are stored.
+ * The tileset is expected to reside at "baseUrl/tileset.json" and to have been produced by the [mesh export service](https://developer.bentley.com/apis/mesh-export/).
+ * If no such tileset exists for the given iModel, return `undefined`.
+ * @see [[FrontendTilesOptions.computeSpatialTilesetBaseUrl]].
+ * @beta
+ */
+export type ComputeSpatialTilesetBaseUrl = (iModel: IModelConnection) => Promise<URL | undefined>;
```

Comment:
After looking in to how the pineapple viewer uses the mesh export service, I decided to remove the `MeshExportServiceProvider` class which was wrapping all of this functionality. The viewer accesses `obtainMeshExportTilesetUrl` directly, so wrapping it in a class would be a breaking change for them.

 Now, the functionality for the mesh export service itself has been moved here, into `MeshExportService.ts` while tests for the service have been moved to `MeshExportService.test.ts`. FrontendTiles functionality has been left in `FrontendTiles.ts` and tests of `initializeFrontendTiles` have been added to `FrontendTiles.test.ts`.